Add support for custom metrics via config

samiwelthomasHO opened the pull request:
Fixes #266

michaelpearsonHO reviewed (changes requested):
`custom-metric-types.libsonnet` currently needs to be bundled within the container at build time. Request changes to look at how a user would mount this file at runtime.

samiwelthomasHO commented on contributing.md:
```diff
@@ -311,7 +311,27 @@ type. If this is the case add the detail dashboard elements keyword to the eleme
+Please refer to the [relevant sections of this documentation](https://github.com/HO-CTO/sre-monitoring-as-code/blob/main/contributing.md#metric-typeslibsonnet) for more information about specifying these
+configuration blocks.
+
+In order for custom metrics to be displayed appropriately on the detail dashboard, it may be necessry to select appropriate dashboard elements to display. There is a `customMetric` set of elements defined in `mac-config.libsonnet` which provides a basic view of custom metric types.
```
Fixed

samiwelthomasHO commented:
@michaelpearsonHO

@arifulhaqueHO and I have taken another look at this and we believe that the custom-metric-types.libsonnet file can now be supplied at runtime, rather than build time.

We've introduced a `.dockerignore` file to prevent the `custom-metric-types.libsonnet` file from being copied into the container image.

We've also modified the `run-mixin.sh` script to check for the presence of a `custom-metric-types.libsonnet` file in the volume mount, and conditionally import that at runtime using an external variable.

Let us know if you have any additional feedback. Thanks.

mahrufiqbalHO reviewed (approved):
Looks good, maybe we can include multiple custom metrics files?
